e2e_tests/tests/cluster/test_checkpoints.py

```diff
@@ -121,6 +120,7 @@ def test_gc_checkpoints_lfs() -> None:
 
 
 @pytest.mark.e2e_cpu
+@pytest.mark.e2e_slurm
```

Comment:
I'm pretty open to not even testing checkpoint gc on slurm.

afaict the only way it actually fails is if bind mounts breaks, and that seems like we probably test it sufficiently elsewhere.

But I added the test because we don't have a good way to expose the gc failure in general, so having one explicit test seems like the right choice.